api/utils/keys/yubikey.go

```diff
@@ -621,15 +665,15 @@ func (y *YubiKey) open() (yk *piv.YubiKey, err error) {
 		Max: time.Millisecond * 50,
 	})
 	if err != nil {
-		return nil, trace.Wrap(err)
+		return trace.Wrap(err)
 	}
 
-	// Backoff and retry for up to 1 second.
-	retryCtx, cancel := context.WithTimeout(context.Background(), time.Second)
+	// Backoff and retry for a time slightly greater than releaseConnectionDelay.
+	retryCtx, cancel := context.WithTimeout(context.Background(), releaseConnectionDelay+100*time.Millisecond)
```

Comment:
I believe this timeout needs to greater than the timeout for releasing a connection. Otherwise, a caller that doesn't use the shared connection has basically no chance to do it.

Unfortunately, this also made the tests take much longer (6-11 seconds for a test). I tried to reuse the same `YubiKey` object as much as possible in the tests, but `keys.GetYubiKeyPrivateKey()` opens its own connection, and I'm not able to easily pass the key that I get from `keys.FindYubiKey()`.